File: Part 2 - Regression/Section 6 - Polynomial Regression/myfiles/regression_template.py
#!/usr/bin/env python2
# -*- coding: utf-8 -*-
"""
Created on Wed May 16 16:03:36 2018

@author: dlog
"""
# Aim:
# General template for Regression models

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd


#####################################################################
# import and split data
#####################################################################
dataset = pd.read_csv('../somefile.csv')
X = dataset.iloc[:,1:2].values # independent vars
Y = dataset.iloc[:,2].values # dependent vars


#####################################################################
# splitting data to training and test sets
#####################################################################
# from sklearn.cross_validation import train_test_split
# X_train, X_test, Y_train, Y_test = train_test_split(X,Y, test_size = 1.0/3.0, random_state = 0)


#####################################################################
# data/feature scaling
#####################################################################
# from sklearn.preprocessing import StandardScaler
# sc_X = StandardScaler()
# X_train = sc_X.fit_transform(X_train)
# X_test = sc_X.transform(X_test)
# sc_Y = StandardScaler()
# Y_train = sc_Y.fit_transform(Y_train)


#####################################################################
# fit Regression model
#####################################################################
#...


#####################################################################
# predict new result 
#####################################################################
Y_pred = regressor.predict(6.5)


#####################################################################
# plots
#####################################################################

plt.scatter(X, Y, color = 'green')
plt.plot(X, regressor.predict(X), color = 'orange')
plt.title('Regression Model')
plt.xlabel('...')
plt.ylabel('...')
plt.show()


# Plot curve with higher/smoother resolution
X_grid = np.arange(min(X), max(X), 0.1)
X_grid = X_grid.reshape((len(X_grid), 1))
plt.scatter(X, Y, color = 'green')
plt.plot(X_grid, regressor.predict(X_grid), color = 'orange')
plt.title('Regression Model')
plt.xlabel('...')
plt.ylabel('...')
plt.show()

File: Part 2 - Regression/Section 6 - Polynomial Regression/myfiles/regression_template.R
# author: dlog
# created: 22 May 2018

# Aim:
# general regression template


#####################################################################
# import csv located one directory below
#####################################################################
dataset = read.csv('../Position_Salaries.csv')
dataset = dataset[,c(2,3)] # keep only Level and Salary cols

#####################################################################
# fit regression model
#####################################################################
# create a regressor here


#####################################################################
# predict new result
#####################################################################
y_pred = predict(regressor, data.frame(Level = 6.5))
y_pred

#####################################################################
# plots
#####################################################################
library(ggplot2)
# normal resolution
ggplot() +
    geom_point(aes(x = dataset$Level, y = dataset$Salary, shape = '...')) +
    geom_line(aes(x = dataset$Level, y = predict(regressor, newdata = dataset), colour = '...')) +
    ggtitle('Regression model') +
    ylab('...') +
    xlab('...') +
    # customise legend titles and shapes
    guides(shape = guide_legend(title = 'Real data', override.aes = list(linetype = 0)), 
           color = guide_legend(title = 'Fitted models'))
ggsave(' .jpg', width = 9, height = 4, units = "cm", scale = 2.5, dpi = 100)


# finer resolution
x_grid = seq(min(dataset$Level), max(dataset$Level), 0.1)
ggplot() +
    geom_point(aes(x = x_grid, y = dataset$Salary, shape = '...')) +
    geom_line(aes(x = x_grid, y = predict(regressor, newdata = data.frame(levels = x_grid)), colour = '...')) +
    ggtitle('Regression model') +
    ylab('...') +
    xlab('...') +
    # customise legend titles and shapes
    guides(shape = guide_legend(title = 'Real data', override.aes = list(linetype = 0)), 
           color = guide_legend(title = 'Fitted models'))
ggsave(' .jpg', width = 9, height = 4, units = "cm", scale = 2.5, dpi = 100)

